### PR TITLE
SAR-10678 Change CVE journey to include 'sign up complete' confirmati…

### PR DESCRIPTION
**Pull request by EviNikolaidou**
…on page

### CODE REVIEW

File: app/uk/gov/hmrc/claimvatenrolmentfrontend/views/sign_up_complete_page.scala.html [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        h1: h1
+)
+
+@()(implicit request: Request[_], messages: Messages, appConfig: AppConfig)
+
+@layout(Some(titleNoForm(messages("sign_up_complete.title")))) {
+
+@h1(messages("sign_up_complete.heading"))
+
+    <p class="govuk-body">@Messages("sign_up_complete.line_1")</p>
```
Comment (HarrNM) [inline app/uk/gov/hmrc/claimvatenrolmentfrontend/views/sign_up_complete_page.scala.html:32]: I think there are paragraph helpers, could we use that here instead, e.g. @p(Html(messages("sign_up_complete.line_1")))

File: app/uk/gov/hmrc/claimvatenrolmentfrontend/controllers/SignUpCompleteController.scala [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+package uk.gov.hmrc.claimvatenrolmentfrontend.controllers
+
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.auth.core.retrieve.v2.Retrievals.internalId
+import uk.gov.hmrc.auth.core.{AuthConnector, AuthorisedFunctions}
+import uk.gov.hmrc.claimvatenrolmentfrontend.config.AppConfig
+import uk.gov.hmrc.claimvatenrolmentfrontend.services.JourneyService
+import uk.gov.hmrc.claimvatenrolmentfrontend.views.html.sign_up_complete_page
+import uk.gov.hmrc.http.InternalServerException
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
```
Comment (HarrNM) [inline app/uk/gov/hmrc/claimvatenrolmentfrontend/controllers/SignUpCompleteController.scala:26]: There's a few unused imports in here

File: it/uk/gov/hmrc/claimvatenrolmentfrontend/views/SignUpCompleteViewTests.scala [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.claimvatenrolmentfrontend.views
+
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import play.api.libs.ws.WSResponse
+import uk.gov.hmrc.claimvatenrolmentfrontend.assets.MessageLookup.{Base, BetaBanner, Header, SignUpCompleteClient => messages}
+import uk.gov.hmrc.claimvatenrolmentfrontend.controllers.routes
```
Comment (HarrNM) [inline it/uk/gov/hmrc/claimvatenrolmentfrontend/views/SignUpCompleteViewTests.scala:23]: Unused import

File: app/uk/gov/hmrc/claimvatenrolmentfrontend/config/AppConfig.scala [hunk 4]
```diff
@@ -50,6 +50,8 @@ class AppConfig @Inject()(config: Configuration, servicesConfig: ServicesConfig)
 
   lazy val btaUrl: String = servicesConfig.getString("bta.url")
```
Comment (HarrNM) [inline app/uk/gov/hmrc/claimvatenrolmentfrontend/config/AppConfig.scala:51]: It looks like on VSU in the SignUpComplete view we use this btaUrl not the relative one added

File: app/uk/gov/hmrc/claimvatenrolmentfrontend/config/AppConfig.scala [hunk 5]
```diff
@@ -50,6 +50,8 @@ class AppConfig @Inject()(config: Configuration, servicesConfig: ServicesConfig)
...
 
+  lazy val businessAccountUrl: String = "/business-account"
```
Comment (woollett-r) [inline app/uk/gov/hmrc/claimvatenrolmentfrontend/config/AppConfig.scala:53]: If we follow the pattern in vat-sign-up "businessAccountUrl" is not required.

File: app/uk/gov/hmrc/claimvatenrolmentfrontend/controllers/CheckYourAnswersController.scala [hunk 6]
```diff
@@ -60,7 +60,7 @@ class CheckYourAnswersController @Inject()(mcc: MessagesControllerComponents,
         case Some(Credentials(credentialId, "GovernmentGateway")) ~ Some(groupId) ~ Some(internalId) =>
           claimVatEnrolmentService.claimVatEnrolment(credentialId, groupId, internalId, journeyId).map {
             case Right(continueUrl) =>
```
Comment (woollett-r) [inline app/uk/gov/hmrc/claimvatenrolmentfrontend/controllers/CheckYourAnswersController.scala:62]: As we no longer use "continueUrl" you can use "case Right(_) =>".

File: app/uk/gov/hmrc/claimvatenrolmentfrontend/controllers/SignUpCompleteController.scala [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+                                         view: sign_up_complete_page,
+                                         journeyService: JourneyService,
+                                         val authConnector: AuthConnector
+                                        )(implicit val config: AppConfig,  ec: ExecutionContext) extends FrontendController(mcc) with AuthorisedFunctions {
+
+
+  def signUpComplete(journeyId:String): Action[AnyContent] = Action.async {
+    implicit request =>
+      authorised().retrieve(internalId) {
+        case Some(authId) =>
```
Comment (woollett-r) [inline app/uk/gov/hmrc/claimvatenrolmentfrontend/controllers/SignUpCompleteController.scala:43]: If we are not using the journey configuration there is no need to use "journeyService.retrieveJourneyConfig". We can remove the service from the class and just use

case Some(authId) => Future.successful(Ok(view())

File: conf/application.conf [hunk 8]
```diff
@@ -109,6 +109,11 @@ microservice {
       host = localhost
       port = 7775
     }
+
```
Comment (woollett-r) [inline conf/application.conf:112]: If we just want the acceptance/performance tests to work we are fine with using the bta.url property. However, this isn't doing any harm.

**Review by woollett-r — APPROVED** [review]

**Pull request merged**